src/esp/bindings/bindings.cpp

```diff
@@ -25,20 +26,47 @@ using namespace py::literals;
 
 #ifdef ESP_BUILD_WITH_CUDA
 #include "esp/sensor/RedwoodNoiseModel.h"
+using esp::sensor::RedwoodNoiseModelGPUImpl;
 #endif
 
 #include <Magnum/ImageView.h>
 #include <Magnum/Python.h>
 #include <Magnum/SceneGraph/Python.h>
 
-using namespace esp;
-using namespace esp::core;
-using namespace esp::geo;
-using namespace esp::gfx;
-using namespace esp::nav;
-using namespace esp::scene;
-using namespace esp::sensor;
-using namespace esp::physics;
+namespace scene = esp::scene;
+using esp::box3f;
+using esp::quatf;
+using esp::vec3f;
+using esp::vec4f;
+using esp::core::Configuration;
+using esp::geo::OBB;
+using esp::gfx::RenderCamera;
+using esp::gfx::Renderer;
+using esp::gfx::RenderTarget;
+using esp::gfx::Simulator;
+using esp::gfx::SimulatorConfiguration;
+using esp::physics::MotionType;
+using esp::scene::Mp3dObjectCategory;
+using esp::scene::Mp3dRegionCategory;
+using esp::scene::ObjectControls;
+using esp::scene::SceneConfiguration;
+using esp::scene::SceneNode;
+using esp::scene::SceneNodeType;
+using esp::scene::SemanticCategory;
+using esp::scene::SemanticLevel;
+using esp::scene::SemanticObject;
+using esp::scene::SemanticRegion;
+using esp::scene::SemanticScene;
+using esp::scene::SuncgObjectCategory;
+using esp::scene::SuncgRegionCategory;
+using esp::scene::SuncgSemanticObject;
+using esp::scene::SuncgSemanticRegion;
+using esp::sensor::Observation;
+using esp::sensor::PinholeCamera;
+using esp::sensor::Sensor;
+using esp::sensor::SensorSpec;
+using esp::sensor::SensorSuite;
+using esp::sensor::SensorType;
```

Comment:
Do we need this, considering we scatter the bindings across multiple modules again on the Python side? To make it easier on both sides, I would say best would be to have `gfx`, `physics`, `scene`, ... submodules generated in the bindings (without using `using`, so it's clear through which submodule should given API be exposed) and then making use of those on the `Python` side instead of have things manually cherry-picked via `__all__`.

Of course there's a question whether this should be done as a part of this PR or separately, but based on previous experience the tech debt TODOs tend to stay in the code forever and so I'd say it should be in this PR. After all, the purpose of this PR is to fix the root cause of the Clang Tidy warnings, not merely silencing them, right? :)